### ISSUE TITLE
Feedback: Animation playback support

### ISSUE THREAD

**Issue by JustinXinLiu** [Microsoft/WindowsUIDevLabs#20]
Sometimes when I define a `KeyFrameAnimation`, I don't just want to play it from beginning to end but to *scrub it through* based on external input.

I am really just looking for something similar to the [`Seek`](https://msdn.microsoft.com/en-us/library/system.windows.media.animation.storyboard.seek(v=vs.110).aspx) function of the `Storyboard`. Also, Web Animations API has [`currentTime`](https://developer.mozilla.org/en-US/docs/Web/API/Animation/currentTime) to do the same thing.

A common scenario would be:
* Define an animation.
* `Begin` and immediately `Suspend` the animation when a finger is down.
* Call animation`.Seek` based on how far the finger has moved from point A to point B.
* Once the finger lifts up, call animation`.Resume` to finish off the animation smoothly.

Maybe this method should be living inside `CompositionScopedBatch` which would allow scrubbing a collection of animations?

**Comment by tomzorz** [comment]
I was just thinking about a similar thing - binding some kind of input to an expression parameter. We can do scroll right now sure... but other things, like: touch/pointerdown duration, position.

Also using an expression animation to bind to property animated by a keyframe one is a nice trick.

**Comment by JustinXinLiu** [comment]
@tomzorz Yeah more support to expression animations is always welcome. :) 

But in my particular case, a `Seek` method would allow me to create *one single* animation that does everything (i.e. starting, scrubbing and resuming an animation). With expression animations, I don't think it could be as simple.

**Comment by pgills** [comment]
I believe you could achieve something like this today by using a combination of `CompositionPropertySet`, `SetReferenceParameter`, and using an `ExpressionKeyframe` for the driving, then pay a separate `KeyframeAnimation` once you've detected `PointerUp`. 

However, today the input capture would happen on the UI thread, so it wouldn't be seamless and buttery smooth.

In terms of "input driven animations," leveraging `ElementCompositionPreview.GetScrollViewerManipulationPropertySet` with an 'ExpressionAnimation' is the closest we have today. You can see that technique in our [Z-OrderScrolling Sample](https://github.com/Microsoft/composition/tree/master/SDK10586_NOV_UPDATE_RTM/Z-OrderScrolling) which I believe you folks have probably already seen.

But it sounds like you'd like some more control @JustinXinLiu and @tomzorz in terms of what the "source input" is, then tie it to an ExpressionAnimation. 

So just to recap the scenario your trying to achieve:
- Define an Animation (let's propose an `ExpressionAnimation`)
- Associate some kind of driving source to the Animation
- Associate the Animation to some property of a `CompositionObject` (think `Visual.Offset` or a property on an `CompositionEffectBrush`)
- Provide some kind of "ending" after you're done with the "driving" animation

In plain words, something like dragging a visual (sticks to your finger) then flicking it, and it lands there with some kind of easing.

**Comment by JustinXinLiu** [comment]
Thanks @pgills yes what you proposed would work, and along with @tomzorz's suggestion to add *source input* would be awesome. 

But what I am proposing is really just how the web animations and the traditional xaml animations control the animation playback. Please have a look at this [demo](https://github.com/JustinXinLiu/AnimationPlayback_XamlStoryboard) that I just wrote and hopefully it could make my point clearer.

IMHO, this is way much simpler than having a lot of calculations inside expression animations or keyframe animations with expression keyframes.

**Comment by pgills** [comment]
@JustinXinLiu that's good feedback. We'll pass this feedback on to the animations and expression crew. 

Here's what I'm hearing:
- @tomzorz: It would be great to have other forms of input drive expressions, not just scrolling today
- @JustinXinLiu: It would be great if I could have more control animation playback that feels similar to something like XAML Storyboards today

I'll leave the issue open for a couple days if you folks have any more feedback in this area. 

Thanks for the feedback the team appreciates it!

**Comment by JustinXinLiu** [comment]
Thanks @pgills! The Composition API is amazing, and I know you guys will have more interesting stuff to share (yeah I already had a look at the new API in *14271* :P), please keep up the good work! :)

**Comment by pgills** [comment]
Thanks!

**Comment by JustinXinLiu** [comment]
I am closing this issue 'cause the new `InteractionTracker` does exactly what I want, actually it does so much more. I cannot wait to try some of the new stuff.

Kudos to the team! :+1:
